Wrapping errors in Node.js using _node-verror_
`process.nextTick()` callback pattern
`try-catch` only for **sync** code
Middleware/pipeline design pattern
Node and `sudo` 